src/Calculator/App.xaml.cpp

```diff
@@ -70,6 +70,33 @@ App::App()
     this->HighContrastAdjustment = ApplicationHighContrastAdjustment::None;
 
     this->Suspending += ref new SuspendingEventHandler(this, &App::OnSuspending);
+    auto value = Windows::Storage::ApplicationData::Current->LocalSettings->Values->Lookup(L"themeSetting");
+    auto DefaultTheme = ref new Windows::UI::ViewManagement::UISettings();
+    auto uiTheme = DefaultTheme->GetColorValue(Windows::UI::ViewManagement::UIColorType::Background).ToString();
+    if (value != nullptr)
+    {
+        String ^ colorSetting = safe_cast<String ^>(value);
+        // Apply theme choice.
+        if (colorSetting == L"Dark")
+        {
+            App::Current->RequestedTheme = ApplicationTheme::Dark;
+        }
+        else if (colorSetting == L"Light")
+        {
+            App::Current->RequestedTheme = ApplicationTheme::Light;
+        }
+        else if (colorSetting == L"System")
```

Comment:
You don't need to explicitly detect and set the system theme. You just need to remove the saved theme from the app local settings. When the app restarts, then it will see there is no saved theme and the UWP defaults will kick in.

Reply:
@sanderl How can I do that? Thank you.

Reply:
If the colorSetting == L"System" you can just not set the App::Current->RequestedTheme and the app will automatically default to the desired system theme.